src/Storages/StorageS3Settings.h

```diff
@@ -21,58 +21,96 @@ namespace DB
 {
 
 struct Settings;
+class NamedCollection;
 
 struct S3Settings
 {
     struct RequestSettings
     {
-        size_t max_single_read_retries = 0;
-        size_t min_upload_part_size = 0;
-        size_t max_upload_part_size = 0;
-        size_t upload_part_size_multiply_factor = 0;
-        size_t upload_part_size_multiply_parts_count_threshold = 0;
-        size_t max_part_number = 0;
-        size_t max_single_part_upload_size = 0;
-        size_t max_single_operation_copy_size = 0;
-        size_t max_connections = 0;
+        struct PartUploadSettings
+        {
+            size_t min_upload_part_size = 16 * 1024 * 1024;
+            size_t max_upload_part_size = 5ULL * 1024 * 1024 * 1024;
+            size_t upload_part_size_multiply_factor = 2;
+            size_t upload_part_size_multiply_parts_count_threshold = 500;
+            size_t max_part_number = 10000;
+            size_t max_single_part_upload_size = 32 * 1024 * 1024;
+            size_t max_single_operation_copy_size = 5ULL * 1024 * 1024 * 1024;
+
+            inline bool operator==(const PartUploadSettings & other) const
```

Comment:
(cosmetic: `inline` is redundant for in-class methods. In fact, you could simply have it default defined: https://stackoverflow.com/a/27837789)